xmtp_mls/src/groups/mls_sync.rs

```diff
@@ -1237,18 +1236,59 @@ where
         envelope: &GroupMessageV1,
         trust_message_order: bool,
     ) -> Result<MessageIdentifier, GroupMessageProcessingError> {
+        let mls_message_in = MlsMessageIn::tls_deserialize_exact(&envelope.data)?;
+        let message_entity_kind = match mls_message_in.wire_format() {
+            WireFormat::Welcome => EntityKind::Welcome,
+            _ => EntityKind::Group,
+        };
```

Comment:
This code block was pulled up from `process_message_inner`, which you can see is the first function called by this method in the old code.

You can see that this same code was pulled down from `consume_message()`, which is one of the callers of this method.